test/scripts/test_valhalla_build_config.py

```diff
@@ -0,0 +1,87 @@
+import unittest
+from unittest.mock import MagicMock
+
+try:
+    from test.scripts import helper
+except (ModuleNotFoundError, ImportError):
+    import helper
+
+# get the module to test
+valhalla_build_config = helper.import_module('valhalla_build_config')
+# mock the add_argument so we can track which arguments it's called with in the main code
+mock_parser = valhalla_build_config.parser
+mock_parser.add_argument = MagicMock()
+
+
+class TestBuildConfig(unittest.TestCase):
+    def test_add_leaf_types(self):
+        help_text = {
+            'bool': "boolean",
+            "str": "string",
+            "list": "list",
+            "opt_str": "optional string",
+            "opt_list": "optional list",
+            "opt_int": "optional int",
+        }
+        config = {
+            'bool': True,
+            'str': 'string',
+            'list': [0, 1, 2],
+            'opt_str': valhalla_build_config.Optional(str),
+            'opt_list': valhalla_build_config.Optional(list),
+            'opt_int': valhalla_build_config.Optional(int),
+        }
+
+        value_types = {
+            "bool": valhalla_build_config.parse_bool,
+            "str": str,
+            "list": valhalla_build_config.comma_separated_list,
+            "opt_str": str,
+            "opt_list": list,
+            "opt_int": int
+        }
+
+        leaves = list()
+        valhalla_build_config.add_leaf_args('', config, leaves, mock_parser, help_text)
+
+        # make sure the parser.add_argument was called with the right stuff while parsing the config
+        for name, default in config.items():
+            mock_parser.add_argument.assert_any_call(
+                f'--{name.replace("_", "-")}', type=value_types[name], help=help_text[name], default=default
+            )
```

Comment:
testing if `parser.add_argument()` gets  called with the right arguments for each of the config members